### ISSUE TITLE
Merge eclipse.platform.ua repo here

### ISSUE THREAD

**Issue by akurtakov** [eclipse-platform/eclipse.platform#324]
Today, I've just fixed the tests/compile on jenkins for platform.ua repo so the repo is open for merging and this one makes most sense to me. Is there anyone up for doing it so I can continue stabilizing other tests?

**Comment by akurtakov** [comment]
@vogella @laeubi maybe you?

**Comment by laeubi** [comment]
I can probably give it a try the next days just in case someone else like to try it in the meantime:

- `git remote add merge-remote <url>` (url = the repo to be merged in the current one)
- `git fetch merge-remote`
- `git merge --allow-unrelated-histories --no-commit --strategy-option ours merge-remote/master`
-` git remote remove merge-remote`
- now enter a nice commit message (e.g. merging <url> into this repository) and commit
- `mkdir <name of merged repo>`
- Then for each folder in the merged repo do `git mv <foldername> <name of merged repo>`
- now enter a nice commit message (e.g moving content of <merged> into subfolder) and commit
- Now usually one needs to adjust the root pom to include the new module folder and create a `pom.xml` in the subfolder that references the root pom as parent and could otherwise be a copy of the original merged repo pom.

**Comment by vogella** [comment]
What is the right build command for both?

platform Jenkins file:
````
steps {
				wrap([$class: 'Xvnc', useXauthority: true]) {
					sh """
					mvn clean verify --batch-mode --fail-at-end -Dmaven.repo.local=$WORKSPACE/.m2/repository \
						-Pbuild-individual-bundles -Pbree-libs -Papi-check \
						-Dcompare-version-with-baselines.skip=false \
						-Dmaven.compiler.failOnWarning=true -Dproject.build.sourceEncoding=UTF-8 -T1C \
						-Dorg.slf4j.simpleLogger.showDateTime=true -Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss.SSS \
						-DtrimStackTrace=false 
					"""
				}
			}
			post {
				always {
					archiveArtifacts artifacts: '.*log,*/target/work/data/.metadata/.*log,*/tests/target/work/data/.metadata/.*log,apiAnalyzer-workspace/.metadata/.*log', allowEmptyArchive: true
					junit '**/target/surefire-reports/TEST-*.xml'
					discoverGitReferenceBuild referenceJob: 'eclipse.platform/master'
					recordIssues publishAllIssues: true, tools: [java(), mavenConsole(), javaDoc()]
				}
````

platform ua Jenkins file:

````
stage('Build') {
			steps {
				wrap([$class: 'Xvnc', useXauthority: true]) {
					sh """
					mvn clean verify --batch-mode --fail-at-end -Dmaven.repo.local=$WORKSPACE/.m2/repository \
						-Pbuild-individual-bundles -Pbree-libs -Papi-check \
						-DskipTests=false -Dcompare-version-with-baselines.skip=false \
						-Dproject.build.sourceEncoding=UTF-8 \
						-Dbuild.sysclasspath=ignore -Dincludeantruntime=false
					"""
				}
			}
			post {
				always {
					archiveArtifacts artifacts: '*.log,*/target/work/data/.metadata/*.log,*/tests/target/work/data/.metadata/*.log,apiAnalyzer-workspace/.metadata/*.log', allowEmptyArchive: true
					junit '**/target/surefire-reports/TEST-*.xml'
					publishIssues issues:[scanForIssues(tool: java()), scanForIssues(tool: mavenConsole())]
				}
				failure {
					script {
						if (env.BRANCH_NAME == 'master') {
				            emailext body: "Please go to <a href='${BUILD_URL}console'>${BUILD_URL}console</a> and check the build failure.<br><br>",
				            subject: "eclipse.platform.ua  master build - BUILD FAILED", 
				            to: "platform-dev@eclipse.org	",
				            from:"genie.releng@eclipse.org"
			            }
		            }
		        }

````

**Comment by akurtakov** [comment]
I think keeping what's in platform is fine. UA has my unsuccessful effort to send emails on master failures so just ignore for now.

**Comment by vogella** [comment]
What about 					-Dbuild.sysclasspath=ignore -Dincludeantruntime=false
?

**Comment by akurtakov** [comment]
Just ignore, shouldn't be needed.

**Comment by vogella** [comment]
Local build is running with mvn clean verify -Pbuild-individual-bundles after the merge. I checked the history for around 10 files and it seems intact. Once the local build is successful I update the change so that others can verify it / check that the history is still present. cc @iloveeclipse

**Comment by vogella** [comment]
Local tests are failing with:

[ERROR] Failed to execute goal on project org.eclipse.update.configurator.tests: Could not resolve dependencies for project org.eclipse.platform:org.eclipse.update.configurator.tests:eclipse-test-plugin:0.1.0-SNAPSHOT: Could not find artifact org.eclipse.help:org.eclipse.help:jar:3.9.100-SNAPSHOT in eclipse-hosted (https://repo.eclipse.org/content/repositories/eclipse/) -> [Help 1]

Seems to be a server issue to me, I re-run without executing the tests.

**Comment by laeubi** [comment]
@vogella probably wait for Tycho fix or run build with `-Dtycho.resolver.classic=true` just to make sure there are no obstacles from there.

**Comment by vogella** [comment]
Thanks @laeubi I now get:

````
INFO] Resolving dependencies of MavenProject: org.eclipse.ua:org.eclipse.ua.tests.doc:1.2.0-SNAPSHOT @ /home/vogella/git/eclipse.platform/ua/org.eclipse.ua.tests.doc/pom.xml
[INFO] {osgi.os=linux, osgi.ws=gtk, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.pde.feature.group 3.14.1500.v20230204-1800
[ERROR]   Missing requirement: org.eclipse.jdt.feature.group 3.19.0.v20230204-1955 requires 'org.eclipse.equinox.p2.iu; org.eclipse.ant.ui [3.8.300.v20220718-1722,3.8.300.v20220718-1722]' but it could not be found
[ERROR]   Missing requirement: org.eclipse.jdt.feature.group 3.19.0.v20230205-1800 requires 'org.eclipse.equinox.p2.iu; org.eclipse.ant.ui [3.8.300.v20220718-1722,3.8.300.v20220718-1722]' but it could not be found
[ERROR]   Missing requirement: org.eclipse.jdt.feature.group 3.19.0.v20230206-1800 requires 'org.eclipse.equinox.p2.iu; org.eclipse.ant.ui [3.8.300.v20220718-1722,3.8.300.v20220718-1722]' but it could not be found
````

**Comment by laeubi** [comment]
Looks like a possibly missing version bump somwhere (because of that you get old features and these reference wrong outdated versions), so maybe check the build output for any "replaced with baseline" warnings but diffferent content.

Or you can create a PR and we see if there are more output....

**Comment by vogella** [comment]
https://github.com/eclipse-platform/eclipse.platform/pull/330

**Comment by vogella** [comment]
Multiple build failures and different onces for GH action and Jenkins. Feel free to push directly to the the branch in case you have an idea how to fix these.

**Comment by vogella** [comment]
@laeubi lets do this in the next release, OK? @akurtakov indicated that this might be too late.

**Comment by akurtakov** [comment]
Yes, every merge uncovers issues and it's already Tuesday with number of potentially breaking builds issues like removing tarball generation pending.

**Comment by HeikoKlare** [comment]
Resolved by #555
